### PR TITLE
Added Outletid

### PR DESCRIPTION
**Pull request by Bowside**
- Added ability to specify outletid, this should cater for newer devices that can control multiple switches per device

### CODE REVIEW

File: sonoff.py [hunk 1]
```diff
@@ -51,35 +46,57 @@ def devices(self):
...
+                     'status': i['params']['switch'],
+                    # 'onlineTime': i['onlineTime'], TODO format as YYYY-MM-DD HH:MM:SS with tz
+                    # 'offlineTime': i['offlineTime'
+                    }
+                devices.append(d)
+        self.devices = devices
 
-    def change_device_status(self, deviceid:str, new_status:str):
-        self.status = [stat.get('status') for stat in self.devices if stat.get('deviceid') == deviceid][0]
+    def change_device_status(self, deviceid: str, outletid: int, new_status: str):
```
Comment (nicolas-cho) [inline sonoff.py]: Can you add pls default value to outletid? --> change_device_status(self, deviceid: str, new_status: str, outletid: int=0)

File: sonoff.py [hunk 2]
```diff
@@ -35,14 +31,13 @@ def login(self):
             'password': self.password
         }
         hex_dig = hmac.new(self.appsecret.encode(),
-                                json.dumps(payload).encode(),
-                                digestmod=hashlib.sha256).digest()
+                           json.dumps(payload).encode(),
```
Comment (nicolas-cho) [inline sonoff.py:34]: in general I prefer to separate between code refactoring commits and new features commits as it make future debugging more difficult. In this case you did very small refactoring changes, so it is ok. thanks.

**Review by nicolas-cho — CHANGES_REQUESTED** [review]
Looks great,
I asked for a small change. thx

File: sonoff.py (hunk 2 above)
Comment (Bowside) [reply sonoff.py:34]: I'll keep this in mind for future changes

File: sonoff.py (hunk 1 above)
Comment (Bowside) [reply sonoff.py]: Good as done

**Review by devnicolasc — APPROVED** [review]
LGTM, Thanks

**Pull request merged**